### ISSUE TITLE
Erasing a package results in the installation of the same package in a different architecture

### ISSUE THREAD

**Issue by jan-kolarik** [openSUSE/libsolv#528]
Hello, I'd like to discuss the behavior that resulted from the recent [commit](https://github.com/openSUSE/libsolv/commit/634371f143e5c0d4244dd46084d0aa7ccdb3967c)'s changes.

Before that, the following test case was passing:
```
repo system 0 testtags <inline>
#>=Pkg: flying 2 9 noarch
#>=Req: P-lib >= 3
#>=Pkg: penny-lib 4 1 x86_64
#>=Prv: P-lib = 3-3

repo main 0 testtags <inline>
#>=Pkg: penny-lib 4 1 x86_64
#>=Prv: P-lib = 3-3
#>=Pkg: penny-lib 4 1 i686
#>=Prv: P-lib = 3-3

system x86_64 rpm system
poolflags implicitobsoleteusescolors

job erase pkg penny-lib-4-1.x86_64@system

result transaction,problems <inline>
#>problem e246fa88 info package flying-2-9.noarch requires P-lib >= 3, but none of the providers can be installed
#>problem e246fa88 solution 1564ffba allow penny-lib-4-1.i686@main
#>problem e246fa88 solution 6c8f7831 erase flying-2-9.noarch@system
#>problem e246fa88 solution bb52fa28 deljob erase pkg penny-lib-4-1.x86_64@system
```
After this change, there is this difference:
```
Results differ:
+erase penny-lib-4-1.x86_64@system
+install penny-lib-4-1.i686@main
...
```
It does not seem right to me, what do you think?

**Comment by mlschroe** [comment]
Hey, don't make this so complicated!

Fixed with commit 5dc4f81fda718df26f8f5dda343a71d34cf555b4

**Comment by j-mracek** [comment]
Thank you  very much for the patch
